hpx/components/containers/vector/vector_segmented_iterator.hpp

```diff
@@ -99,6 +101,46 @@ namespace hpx
         boost::shared_ptr<server::partition_vector<T> > data_;
     };
 
+    template <typename T, typename BaseIter>
+    class const_local_raw_vector_iterator
+      : public boost::iterator_adaptor<
+            const_local_raw_vector_iterator<T, BaseIter>, BaseIter
+        >
+    {
+    private:
+        typedef boost::iterator_adaptor<
+                const_local_raw_vector_iterator<T, BaseIter>, BaseIter
+            > base_type;
+        typedef BaseIter base_iterator;
+
+    public:
+        typedef const_local_vector_iterator<T> local_iterator;
+        typedef const_local_vector_iterator<T> local_const_iterator;
+
+        const_local_raw_vector_iterator(base_iterator const& it,
+                boost::shared_ptr<server::partition_vector<T> > const& data)
+          : base_type(it), data_(data)
+        {}
+
+        local_iterator remote()
```

Comment:
This looks odd: The const iterator returning a non-const iterator to its local counterpart.